ledger/store/src/block/mod.rs

```diff
@@ -772,20 +772,9 @@ pub trait BlockStorage<N: Network>: 'static + Clone + Send + Sync {
         }
     }
 
-    /// Returns the block that contains the specified certificate (if any).
-    fn get_block_for_certificate(&self, certificate_id: &Field<N>) -> Result<Option<Block<N>>> {
-        // Retrieve the height and round for the given certificate ID.
-        let (block_height, _) = match self.certificate_map().get_confirmed(certificate_id)? {
-            Some(res) => cow_to_copied!(res),
-            None => return Ok(None),
-        };
-
-        // Retrieve the block hash.
-        let Some(block_hash) = self.get_block_hash(block_height)? else {
-            bail!("The block hash for block '{block_height}' is missing in block storage")
-        };
-        // Retrieve the block.
-        self.get_block(&block_hash)
+    /// Returns true if there is a block that contains the specified certificate.
+    fn has_block_for_certificate(&self, certificate_id: &Field<N>) -> Result<bool> {
```

Comment:
We try to use `contains_*` for these types of functions
```suggestion
    fn contains_block_for_certificate(&self, certificate_id: &Field<N>) -> Result<bool> {
```